lib/ain-grpc/src/call_request.rs

```diff
@@ -39,6 +39,25 @@ pub struct CallRequest {
 }
 
 impl CallRequest {
+    pub fn guess_tx_type(&mut self) -> Result<(), Error> {
```

Comment:
What's the context here ? Why do we want to re-assign to transaction_type ?

Reply:
@shohamc1 I guess the tradeoff in abstracting this out of th effective_gas_price calculation is that we cannot keep CallRequest immutable. Not sure if that is the best way although i think the idea is nice, to set a tx type. We could abstract this into our custom deserialize implementation actually which may keep it cleaner. Alternatively not sure if abstracting this here is cleaner - maybe keeping it inside effective_gas_price calculation, since the tx type in the call context is really only relevant for gas price calculations, may be safer.

Reply:
Made `guess_tx_type` not mutate `CallRequest` in 506043e as a temporary fix to unblock RPCs. We can merge this and then explore what the best way to handle `transaction_type` will be.